common4j/src/main/com/microsoft/identity/common/java/telemetry/Telemetry.java

```diff
@@ -272,7 +277,9 @@ public void flush(@NonNull final String correlationId) {
         }
 
         for (@SuppressWarnings(WarningType.rawtype_warning) ITelemetryObserver observer : mObservers) {
-            if (observer instanceof ITelemetryAggregatedObserver) {
+            if (observer instanceof IBrokerTelemetryObserver) {
+                new BrokerTelemetryAdapter((IBrokerTelemetryObserver) observer).process(finalRawMap);
+            } else if (observer instanceof ITelemetryAggregatedObserver) {
```

Comment:
@shahzaibj we need to have our own adapter, hence we needed the `IBrokerTelemetryObserver` interface with no methods.

Reply:
We should probably have the observers providing an adapter through an abstract method, in order to get rid of the `if-else` statements, but that may need some refactoring to be done. Probably something we can look at later.

Reply:
Yes we should. Also I see that there is currently no different between TelemetryAggregationAdapter and BrokerTelemetryAdater, so not clear to me why we even need the separate adapter